### ISSUE TITLE
Redis rework

### ISSUE THREAD

**Issue by 0xERR0R** [0xERR0R/blocky#945]
- [ ] Implementation of GroupedCache to enable read access with multiple groups. (#944)
- [ ] Redis implementation of GroupedCache (using [rueidis](https://github.com/rueian/rueidis))
- [ ] Redis implementation of ExpirationCache (DNS response caching) (using [rueidis](https://github.com/rueian/rueidis))
- [ ] Rework Prefetching (Redis & in-memory implementation)
- [ ] if Redis enabled: execution of Jobs (e.g. List refresh, periodic metrics with lists count, prefetching, ...?) only on one instance, (using Redis lock with https://github.com/rueian/rueidis/tree/main/rueidislock)

**Comment by kwitsch** [comment]
I picked this one up again since rueidis is stable now and I hadn't worked with it in a while.

I'm stuck with the unit tests(again). 😞 
Most of the tests need to be rewritten since minredis don't support some features which rueidis neets. 

I can't get my head around how to solve the requirements of the test setups with rueidismock. 

I'm currently considering simply spinning up a test container with redis running alongside the tests...

**Comment by 0xERR0R** [comment]
If we can cover all use cases with e2e tests, why not? Unit test can be helpful to test some edge cases, but if it needs much efford to mock the infrastructure, it is better to throw them away 😄

**Comment by kwitsch** [comment]
I'll open a PR draft without the tests soon.
Maybe I'm only missing something and it isn't that hard.


more eyes & more thoughts may help 😅

**Comment by Serverfrog** [comment]
can this ticket also consider some implementation of #694 ?
As im currently running a kind of GitOps where blocky is also is configured, i don't really want to go my way out to somehow put a Secret i have into a Configmap.
For me the Redis Password is coming from a SecretProvider so its kinda hard to put a singular value from these into a Configmap, and just putting the value itself in there defies the usage of the SecretProvider for it

**Comment by 0xERR0R** [comment]
In your case, you have the password in k8s secret and you can mount it to env variable (for example) and it would be sufficient if blocky can read the password from the env variable?

**Comment by Serverfrog** [comment]
For me personally at my current install it would be not so much a problem, but also consider https://github.com/0xERR0R/blocky/issues/694#issuecomment-1420920276
TL;DR Env Variables are kinda public, in usage of Docker (so in my instance not the case) every user on the machine can read them.
